scripts/maintenance/migrate_project/src/db.py

```diff
@@ -0,0 +1,163 @@
+
+    projects_to_migrate.append(project)
+
+    if hidden_projects_for_user:
+        # extract all hidden projects and check if they require syncing
+        hidden_projects_cursor = _get_hidden_project(src_conn, hidden_projects_for_user)
+        for hidden_result in hidden_projects_cursor:
+            hidden_project = dict(hidden_result.items())
+            if _project_uuid_exists_in_destination(dst_conn, hidden_project["uuid"]):
+                print(f"\n\n>>>>>>>>>SKIPPING, sync for {project=}\n\n")
+                skipped_projects.append(project)
+                continue
+
+            projects_to_migrate.append(hidden_project)
+
+    # check file_meta_data in the projects to migrate
+    for project in projects_to_migrate:
+        user_id = project["prj_owner"]
+        project_id = project["uuid"]
+
+        files_metadata_cursor = _get_file_meta_data_without_soft_links(
+            connection=src_conn, user_id=user_id, project_id=project_id
+        )
+        for result in files_metadata_cursor:
+            file_meta_data = dict(result.items())
+            file_uuid = file_meta_data["file_uuid"]
+
+            if _meta_data_exists_in_destination(dst_conn, file_uuid):
+                print(f"\n\n>>>>>>>>>SKIPPING, sync for {file_meta_data=}\n\n")
+                skipped_files_meta_data.append(file_meta_data)
+                continue
+
+            files_meta_data_to_migrate.append(file_meta_data)
+
+    if len(skipped_projects) > 0:
+        print("SKIPPED projects", len(skipped_projects))
+    if len(skipped_files_meta_data) > 0:
+        print("SKIPPED files", len(skipped_files_meta_data))
+
+    print("Projects to move", len(projects_to_migrate))
+    print("Files to move", len(files_meta_data_to_migrate))
+    print("Projects to SKIP", len(skipped_projects))
+    print("Files to SKIP", len(skipped_files_meta_data))
+
+    # if files and projects already exist
+    if len(skipped_files_meta_data) > 0 or len(skipped_projects) > 0:
+        print(
+            "Projects skipped uuid(primary keys) listing:",
+            [x["uuid"] for x in skipped_projects],
+        )
+        print(
+            "File meta data skipped file_uuid(primary keys) listing:",
+            [x["file_uuid"] for x in skipped_files_meta_data],
+        )
+        raise Exception(
+            "Could not continue migration, some projects or files already exist."
+        )
+
+    return projects_to_migrate, files_meta_data_to_migrate
+
+
+def insert_file_meta_data(connection: Connection, data: Dict[str, Any]) -> None:
+    insert_stmt = insert(file_meta_data).values(**data)
+    on_update_stmt = insert_stmt.on_conflict_do_update(
+        index_elements=[file_meta_data.c.file_uuid], set_=data
+    )
+    connection.execute(on_update_stmt)
+
+
+def insert_projects(connection: Connection, data: Dict[str, Any]) -> None:
+    insert_stmt = insert(projects).values(**data)
+    on_update_stmt = insert_stmt.on_conflict_do_update(
```

Comment:
question, I see above in ```get_project_and_files_to_migrate``` that you check if the projects/file already exist. Why when there is a conflict you do not then skip this?
you could even rollback if necessary. Are you handling that changed uuid? if not I would actually not insert it and stop that project.

Reply:
@mrnicegyu11 replied above

@sanderegg I could do 2 things. `on conflict do nothing` (skipping insertion) or raise an error if there is a conflict.
Which do you thing it would be better.

Reply:
but if you skip just one file, you will have incomplete projects in the destination right?

Reply:
Will raise an error if something happens here.